qiskit/qobj/utils.py

```diff
@@ -1,46 +0,0 @@
-# This code is part of Qiskit.
-#
-# (C) Copyright IBM 2017, 2018.
-#
-# This code is licensed under the Apache License, Version 2.0. You may
-# obtain a copy of this license in the LICENSE.txt file in the root directory
-# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
-#
-# Any modifications or derivative works of this code must retain this
-# copyright notice, and modified files need to carry a notice indicating
-# that they have been altered from the originals.
-
-"""Qobj utilities and enums."""
-
-from enum import Enum, IntEnum
-
-from qiskit.utils import deprecate_func
-
-
-@deprecate_func(
-    since="1.2",
-    removal_timeline="in the 2.0 release",
-    additional_msg="The `Qobj` class and related functionality are part of the deprecated `BackendV1` "
-    "workflow,  and no longer necessary for `BackendV2`. If a user workflow requires `Qobj` it likely "
-    "relies on deprecated functionality and should be updated to use `BackendV2`.",
-)
-class QobjType(str, Enum):
-    """Qobj.type allowed values."""
-
-    QASM = "QASM"
-    PULSE = "PULSE"
-
-
-class MeasReturnType(str, Enum):
-    """PulseQobjConfig meas_return allowed values."""
-
-    AVERAGE = "avg"
-    SINGLE = "single"
-
-
-class MeasLevel(IntEnum):
-    """MeasLevel allowed values."""
-
-    RAW = 0
-    KERNELED = 1
-    CLASSIFIED = 2
```

Comment:
Will a warning about the move of these get into 1.4? It might be worth importing these back here from the new location, maybe with a module level `__getattr__` to warn about the move since it is close to 2.0 and these are referenced in other packages.

Reply:
My comment was made assuming `MeasLevel` and `MeasReturnType` were still public but I see now that they are in `qiskit.result.model` and not added to the docs. Are they not intended to be public any more?

Reply:
Hmm I see. I wasn't aware of the full extent of the use of these classes, and I treated them as an internal migration. At first sight, they were not part of the docs, not even in the original location, and they were buried among other outdated utilities, so I didn't think they would be used externally. Now taking a closer look, they are referenced in other classes, and clearly intended to be public even though they technically didn't have docs entries. 

I can add a warning for the import change in 1.4.1 and properly document them here, to give them official "public" status.

Reply:
Thanks, @ElePT! I missed the fact that this was always internal before with my previous messages :man_facepalming: With the commit you mentioned, you added these classes to `qiskit.result` but didn't add them to the docs. Will they still be considered private going forward?

I checked qiskit-experiments' usage and it only uses `=` and `==` with the enum members so I think we could redefine the enums internally if necessary. I was a little worried about some of the other enum features like `in` where using two definitions of the enum might not work.

Reply:
Yes, I confirmed that they will stay private. I think that vendoring them in `qiskit-experiments` would be a good idea moving forward.